src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -116,8 +116,14 @@ protected void handlePut(RestChannel channel, final RestRequest request, final C
             }
         }
 
-        // if password is set, it takes precedence over hash
+        // Reject empty password
         final String plainTextPassword = securityJsonNode.get("password").asString();
+        if (plainTextPassword != null && plainTextPassword.isEmpty()){
+            badRequestResponse(channel, "Please specify a non-empty 'password' when creating a new internal user.");
```

Comment:
Can you reject in CredentialsValidator which is used by the account API for self-change password as well ?

Reply:
I tried that. But CredentialsValidator validates regex, and it respondes status code 201: {"status":"error","reason":"Password does not match minimum criteria"}. I believe what we want for empty password is status code 400: {"status":"BAD_REQUEST","message":"Please specify a non-empty 'password' when creating a new internal user."}.

Reply:
It returns 201 with that error message? This should not happen. 400 must be returned. Can you verify again

Reply:
Moved the validation to CreadentialValidator.